.changelog/26274.txt

```diff
@@ -0,0 +1,7 @@
+```release-note:new-resource
+aws_dx_macsec_key
```

Comment:
Have you considered naming this resource `aws_dx_macsec_key_association`? Since MACsec keys can only be associated/described/dissociated, renaming the resource may help preventing any confusion. 
```suggestion
aws_dx_macsec_key_association
```

Reply:
Agree that the resource should be named `aws_dx_connection_macsec_key_association`.